src/compiler/checker.ts

```diff
@@ -3427,9 +3427,15 @@ namespace ts {
                 }
 
                 function writeTypeOfSymbol(type: ObjectType, typeFormatFlags?: TypeFormatFlags) {
+                    if (typeFormatFlags & TypeFormatFlags.InArrayType) {

```

Comment:
No; then I parenthesize within unions and intersections, which I want to avoid as it is unneeded.